app/component/itinerary/navigator/hooks/useRealtimeLegs.js

```diff
@@ -149,7 +172,27 @@ const useRealtimeLegs = (initialLegs, mapRef, relayEnvironment) => {
     return () => clearInterval(interval);
   }, [fetchAndSetRealtimeLegs]);
 
-  return { realTimeLegs, time };
+  let currentIndex = -1;
+
+  if (realTimeLegs) {
+    if (time >= legTime(realTimeLegs[realTimeLegs.length - 1].end)) {
+      currentIndex = realTimeLegs.length - 1;
+    } else {
+      currentIndex = realTimeLegs?.findIndex(
+        l => legTime(l.start) <= time && time <= legTime(l.end),
+      );
+    }
+  }
+
+  return {
+    realTimeLegs,
+    time,
+    origin,
+    firstLeg: realTimeLegs?.[0],
+    lastLeg: realTimeLegs?.[realTimeLegs.length - 1],
+    currentLeg: realTimeLegs?.[currentIndex],
+    nextLeg: realTimeLegs?.[currentIndex + 1],
```

Comment:
This is highly suspicious. JS might tolerate `realTimeLegs[undefined]`, but surely not `realTimeLegs[undefined + 1]`. See my comment above about findIndex.